### PR TITLE
Prevents uploading unused vertices to GPU.

### PR DESCRIPTION
**Pull request by olliwang**
This commit fixes the bug that `glnvg__renderFill()` allocates unused quad vertices for `GLNVG_CONVEXFILL`, which causes unused vertices uploaded to the GPU.

Besides, the quad for the `GLNVG_FILL` type is changed to drawn by `GL_TRIANGLE_STRIP` instead of `GL_TRIANGLES` so two more vertices can be reduced per quad.

In one of my test case, I need to paste an image to a framebuffer about 5,000 times, and this change would reduce the total number of uploaded vertices from `52,106` to `22,476`. It is good for FPS and battery consumption.

### CODE REVIEW

File: src/nanovg_gl.h [hunk 1]
```diff
@@ -1354,9 +1354,11 @@ static void glnvg__renderFill(void* uptr, NVGpaint* paint, NVGcompositeOperation
 
 	if (npaths == 1 && paths[0].convex)
 		call->type = GLNVG_CONVEXFILL;
+ 	else
```
Comment (memononen) [inline src/nanovg_gl.h]: This code is a bit hard to follow not. How about:
```C
...
call->type = GLNVG_FILL;
call->triangleCount = 4;
...
if (npaths == 1 && paths[0].convex) {
 	call->type = GLNVG_CONVEXFILL;
 	call->triangleCount = 0;  // Bounding box fill quad not needed for convex fill
}
```

**Comment by olliwang** [comment]
@memononen Thanks for your advise. Just changed it. :)

**Comment by memononen** [comment]
Thanks for the optimization!

**Comment by olliwang** [comment]
You are welcome. 👍
